.github/workflows/reusable-e2e.yml

```diff
@@ -340,7 +347,7 @@ jobs:
 
   e2e-ios:
     needs: check-changed-packages
-    if: ${{ needs.check-changed-packages.outputs.has-changed-packages == 'true' }}
+    if: ${{ always() && (inputs.skip-changed-packages-check || needs.check-changed-packages.outputs.has-changed-packages == 'true') }}
```

Comment:
`always` make sure the job always execute even though its dependent jobs get skipped

Reply:
This is confusing. But more importantly, it seems like `always` makes the job not cancellable, check this [thread](https://github.com/orgs/community/discussions/26303)

Reply:
Another way is we skip `check-changed-packages` at step level without using `always()`.

Reply:
In scenario where `check-changed-packages` was skipped, `e2e-ios` and `e2e-android` would be skipped alongside.

Reply:
Yeah this is tricky. May need to use `!cancel() && !failure()` or Skip at step level can get rid of this spaghetti conditional